### DISCUSSION TITLE
Image qualities

### DISCUSSION THREAD

**Discussion by Ruttie2006** [AvaloniaUI/Avalonia#9291] [Q&A] [answered]
I assumed this was asked before, so I've searched but didn't find anything, so I'll ask myself:

How do you specify different image qualities?
If we take [Uno](https://platform.uno/docs/articles/features/working-with-assets.html#scale) for example, we can simply add `.scale-xxx` to specify different qualities.
Maybe there is an obvious piece of documentation describing this and I've just missed it, but does avalonia have a system to do something like that?

**Comment by maxkatz6** [comment, marked as answer]
Hi! No, it's not supported currently.
But I think it could be a great feature request.
Checking UWP documentation, https://learn.microsoft.com/en-us/windows/uwp/app-resources/tailor-resources-lang-scale-contrast
We should have something at least for:
- scale
- theme
- layout direction

Platform dependent assets can be used with OnPlatform extension (which is in work). 
Language dependent assets are way more complex question, https://github.com/AvaloniaUI/Avalonia/discussions/8741.

**Comment by maxkatz6** [comment]
Also, currently it shouldn't be hard to implement this manually. 
You can create a markup extension, use an attached property, use converter or do it from code.
Main points would be to:
- Get current scaling from window: `window.PlatformImpl.RenderScaling` and subscribe on `window.PlatformImpl.ScalingChanged` to keep it dynamic.
- Format a correct physical path to the asset
- Resolve an asset with
```
var stream = AvaloniaLocator.Current.GetService<IAssetLoader>().Open(new Uri("formatted-asset-link"));
image.Source = new Bitmap(stream);
```

**Reply by Ruttie2006** [reply]
I'll look into it, thank you!
